galsim/detectors.py

```diff
@@ -277,6 +275,29 @@ def applyIPC(self, IPC_kernel, edge_treatment='extend', fill_value=None, kernel_
     else:
         self.array[:,:] = out_array
 
+def quantize(self):
+    """
+    Rounds the pixel values in an image to integer values, while preserving the type of the data.
+
+    At certain stages in the astronomical image generation process, detectors effectively round to
+    the nearest integer.  The exact stage at which this happens depends on the type of device (CCD
+    vs. NIR detector).  For example, for H2RG detectors, quantization happens in two stages: first,
+    when detecting a certain number of photons, corresponding to the sum of background and signal
+    multiplied by the QE and including reciprocity failure.  After this, a number of other processes
+    occur (e.g., nonlinearity, IPC, read noise) that could result in non-integer pixel values, only
+    rounding to an integer at the stage of analog-to-digital conversion.
+
+    Because we cannot guarantee that quantization will always be the last step in the process, the
+    quantize() routine does not actually modify the type of the image to 'int'.  However, users can
+    easily do so by doing
+
+        image.quantize()
+        int_image = galsim.Image(image, dtype=int)
+
+    """
+    self.applyNonlinearity(numpy.round)
```

Comment:
Haha, I'm very good at copying other people's elegant code ideas and using them elsewhere.  :)